docusaurus/docs/dev-docs/configurations/typescript.md

```diff
@@ -16,3 +22,42 @@ description: How to configure Strapi for TypeScript development.
 | `tsconfig.json` file                      | application root | Manages TypeScript compilation for the server.                                                                                                    |
 | `tsconfig.json` file                      | `./src/admin/`   | Manages TypeScript compilation for the admin panel.                                                                                               |
 
+## Strapi-specific configuration for TypeScript
+
+Strapi supports types autocompletion in your code editor while you are developing. Autocompletion works for both JavaScript and TypeScript projects.
```

Comment:
Same as my comment above, maybe we could turn that into smth more like

"while Strapi provides generic typed APIs by default, they can be adapted to reflect the exact state of your application thanks to types auto-generation" (or something like that, you get the idea)

What do you think?